src/hinge.cpp

```diff
@@ -0,0 +1,96 @@
+#include "wayfire/plugin.hpp"
+#include "wayfire/signal-definitions.hpp"
+#include "wayfire/input-device.hpp"
+#include <wayfire/util/log.hpp>
+#include <fstream>
+#include <vector>
+#include <string>
+
+class wayfire_hinge : public wf::plugin_interface_t
+{
+    wf::option_wrapper_t<std::string> file_name{"hinge/filename"};
+    wf::option_wrapper_t<int> poll_freq{"hinge/poll_freq"};
+    wf::option_wrapper_t<int> flip_degree{"hinge/flip_degree"};
+    
+    std::ifstream device_file;
+    wf::wl_timer* timer;
+
+    bool input_enabled = true;
+
+    std::vector<nonstd::observer_ptr<wf::input_device_t>> get_inputs() {
+        wf::compositor_core_t* core = &wf::get_core();
+        return core->get_input_devices();
```

Comment:
nitpick/fyi: you can save the return value like this: `wf::compositor_core_t &core = wf::get_core()`
or even better, just use it directly: `return wf::get_core()->get_input_devices();`